### PR TITLE
Add v4 upgrade doc

### PR DESCRIPTION
**Pull request by Dev25**
Edit as you see fit @morgante 

Follow up from #117

### CODE REVIEW

**Comment by Dev25** [comment]
I've never really used the state migration scripts before, i can take a look but its gonna have to wait until later next week

**Comment by morgante** [comment]
@Dev25 Thanks. Let me know if you have any questions.

File: modules/mysql/read_replica.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 locals {
   replicas = {
-    for x in var.read_replicas : x.name => x
+    for x in var.read_replicas : "${local.master_instance_name}-replica${var.read_replica_name_suffix}${x.name}" => x
```
Comment (Dev25) [inline modules/mysql/read_replica.tf:19]: This was done in order for the migration script to easily work on existing resource attributes. Anyone who's been using `master` directly recently will need to do a custom state migration manually.

**Comment by Dev25** [comment]
@morgante PTAL

I've done a quick test using the mysql module/migrate script but would appreciate if someone else could follow the docs in case i've missed anything.

File: helpers/migrate4.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env python3
```
Comment (Dev25) [inline helpers/migrate4.py:1]: This script is based on the [GKE for_each migrate script ](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/helpers/migrate7.py). The only changes is `MIGRATIONS` list + copyright year bump

**Comment by Dev25** [comment]
Not sure why tests are failing, it's broken on master too. Any chance we can get this released soon? @morgante

**Comment by morgante** [comment]
@Dev25 Sorry for the delay - could you merge in master?

**Comment by Dev25** [comment]
Done! @morgante

File: docs/upgrading_to_sql_db_4.0.0.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+       host     = "localhost"
+     }
+   ]
+
+```
+
+#### Read Replicas
+The new `read_replicas` variable is used to manage all replica configuration. In order to migrate existing replicas you will need to perform a state migration.
+
+- You must have `read_replica_size` objects inside `read_replicas`
```
Comment (syedrakib) [inline docs/upgrading_to_sql_db_4.0.0.md:43]: Shouldn't this be "You must **not** have read_replica_size...." ?

The `read_replicas` array itself would be the size.

**Pull request merged**